### PR TITLE
Use collectAsStateWithLifecycle

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Use collectAsStateWithLifecycle

#### WHY

We have a standard implement now, and the current name scares devrel.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ androidx-wear-watchface = "1.2.0-alpha04"
 androidxActivity = "1.6.1"
 androidxCore = "1.9.0"
 androidxFragment = "1.5.4"
-androidxLifecycle = "2.5.1"
+androidxLifecycle = "2.6.0-alpha03"
```
Comment (ataulm) [inline gradle/libs.versions.toml:14]: What's the impact of depending on an alpha version here for consumers of Horologist? Will such projects automatically use this version even if they directly depended on an older version?

Comment (yschimke) [reply gradle/libs.versions.toml:14]: Yep - I'm torn.  I was staying on releases, but we do use some alphas,betas,rc, so it's not clear it's wrong.  Alphas are high quality.

**Review by astamato — APPROVED** [review]
LGTM if you're ok with the alpha dependency

**Review by ataulm — APPROVED** [review]
:+1: same as Ale - I think it's fine but I wonder if it's unexpected: can we be explicit about what our strategy is going forward so that we don't have to make such decisions on a case by case basis?

**Comment by yschimke** [comment]
https://github.com/google/horologist/issues/736

**Pull request merged**